libafl/src/executors/inprocess.rs

```diff
@@ -1285,12 +1290,34 @@ impl InChildProcessHandlers {
     {
         unsafe {
             let data = &mut FORK_EXECUTOR_GLOBAL_DATA;
-            child_signal_handlers::setup_child_panic_hook::<E, I, OT, S>();
+            // child_signal_handlers::setup_child_panic_hook::<E, I, OT, S>();
+            setup_signal_handler(data)?;
+            compiler_fence(Ordering::SeqCst);
```

Comment:
Are we sure a `compiler_fence` is fine, here? The hardware is still allowed to reorder (however it's only for signal handling, so no threading, right?)
A proper fence may be more sound here

> . Note that it does not prevent the hardware from doing such re-ordering. This is not a problem in a single-threaded, execution context, but when other threads may modify memory at the same time, stronger synchronization primitives such as [fence](https://doc.rust-lang.org/std/sync/atomic/fn.fence.html) are required.

https://doc.rust-lang.org/std/sync/atomic/fn.compiler_fence.html

Reply:
I think it's fine...
it's setting up the signal handler here, and it is called from the construtor of this executor...
after this setup, it would takes a while before the fuzzer enters the harness execution and any signal got delivered.

Reply:
yes